### ISSUE TITLE
use a fast keccak implementation

### ISSUE THREAD

**Issue by banteg** [ApeWorX/ape#874]
### Overview

`eth_utils.keccak` uses the auto backend in `eth-hash`, which defaults to `pycryptodome` even if `pysha3` is installed. we should always choose pysha3, which is 10x faster.

see this issue and run the benchmark to see for yourself https://github.com/ethereum/eth-hash/issues/35

some results from the different machines i run
```python
# 1
eth_hash 2.44878625
web3 4.234222375
cryptodome 4.271701542
pysha3 0.44263583300000064

# 2
eth_hash 4.799470028025098
web3 8.730179763981141
cryptodome 8.537778785103
pysha3 0.8991878429660574
```

### Specification

either define a new utility which uses pysha3 or set `os.environ['ETH_HASH_BACKEND'] = 'pysha3'` _before_ importing `eth_utils`.

### Dependencies

add pysha3 as a requirement

**Comment by banteg** [comment]
maybe the problem will resolve itself with the next release of eth-hash, my pr just got merged
https://github.com/ethereum/eth-hash/pull/43

**Comment by banteg** [comment]
my patch got merged eth-hash 0.5.0, now we just need to wait till it trickles down to eth-utils and web3.py. i'll close this since it's no longer relevant from ape side.
